lib/std/once.nim

```diff
@@ -0,0 +1,31 @@
+import atomics
+import locks
```

Comment:
what about re-entrant code? do we need a `ronce / reentrantOnce`, or can `once` be made re-entrant safe without overhead?